inky/inky_e673.py

```diff
@@ -0,0 +1,363 @@
+"""Inky e-Ink Display Driver."""
+import time
+import warnings
+
+import gpiod
+import gpiodevice
+import numpy
+from gpiod.line import Bias, Direction, Value
+from PIL import Image
+
+from . import eeprom
+
+BLACK = 0
+WHITE = 1
+YELLOW = 2
+RED = 3
+BLUE = 4
+GREEN = 5
+
+DESATURATED_PALETTE = [
+    [0, 0, 0],
+    [255, 255, 255],
+    [255, 255, 0],
+    [255, 0, 0],
+    [0, 0, 255],
+    [0, 255, 0],
+    [255, 255, 255]]
+
+SATURATED_PALETTE = [
+    [0, 0, 0],
+    [161, 164, 165],
+    [208, 190, 71],
+    [156, 72, 75],
+    [61, 59, 94],
+    [58, 91, 70],
+    [255, 255, 255]]
+
+RESET_PIN = 27  # PIN13
+BUSY_PIN = 17   # PIN11
+DC_PIN = 22     # PIN15
+
+MOSI_PIN = 10
+SCLK_PIN = 11
+CS0_PIN = 8
+
+EL673_PSR = 0x00
+EL673_PWR = 0x01
+EL673_POF = 0x02
+EL673_POFS = 0x03
+EL673_PON = 0x04
+EL673_BTST1 = 0x05
+EL673_BTST2 = 0x06
+EL673_DSLP = 0x07
+EL673_BTST3 = 0x08
+EL673_DTM1 = 0x10
+EL673_DSP = 0x11
+EL673_DRF = 0x12
+EL673_PLL = 0x30
+EL673_CDI = 0x50
+EL673_TCON = 0x60
+EL673_TRES = 0x61
+EL673_REV = 0x70
+EL673_VDCS = 0x82
+EL673_PWS = 0xE3
+
+_SPI_CHUNK_SIZE = 4096
+
+_RESOLUTION_7_3_INCH = (800, 480)  # Inky Impression 7.3 (Spectra 6)"
+
+_RESOLUTION = {
+    _RESOLUTION_7_3_INCH: (_RESOLUTION_7_3_INCH[0], _RESOLUTION_7_3_INCH[1], 0, 0, 0, 0b01),
+}
+
+
+class Inky:
+    """Inky e-Ink Display Driver."""
+
+    BLACK = 0
```

Comment:
I think these need to match the weird gappy colours? For users calling set_pixel rather than setting a PIL image.

ie:

```
BLACK = 0
WHITE = 1
YELLOW = 2
RED = 3
BLUE = 5
GREEN = 6
```

Reply:
Hi @Gadgetoid, sorry to barge into the PR like this. I am just wondering what you mean by "gappy colours"?

I am looking to port this driver to .NET and am trying to follow along.

Reply:
Colour "4" doesn't map to a specific colour on the display. I believe it comes out white, but there's no guarantee. It's mostly a concern with our implementation due to the way we handle dithering but the constants need to also reflect that there's no 4. Gappy!

Reply:
Perfect - thanks for the explanation!

Reply:
> Colour "4" doesn't map to a specific colour on the display. I believe it comes out white, but there's no guarantee. It's mostly a concern with our implementation due to the way we handle dithering but the constants need to also reflect that there's no 4. Gappy!

It's orange on my 7.3" from waveshare, but apparently not on the 13.3".
https://github.com/waveshareteam/e-Paper/pull/377